src/lib/util/zippath.cpp

```diff
@@ -111,9 +111,9 @@ int is_path_separator(char c)
  * @param [in,out]  endpos      If non-null, the endpos.
  */
 
-static void parse_parent_path(const char *path, int *beginpos, int *endpos)
+static void parse_parent_path(const std::string &path, std::string::size_type *beginpos, std::string::size_type *endpos)
 {
-	int length = strlen(path);
+	std::string::size_type length = path.length();
 	int pos;
```

Comment:
This variable being a signed integer doesn't play nice with `std::string::size_type` being unsigned. The whole function needs adjusting.